core/parentchain/block-import-dispatcher/src/triggered_dispatcher.rs

```diff
@@ -0,0 +1,232 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! A block import dispatcher that retains all blocks in a queue until import is triggered.
+
+use crate::{
+	error::{Error, Result},
+	DispatchBlockImport,
+};
+use itc_parentchain_block_import_queue::{PopFromBlockQueue, PushToBlockQueue};
+use itc_parentchain_block_importer::ImportParentchainBlocks;
+use std::{sync::Arc, vec::Vec};
+
+/// Trait to specifically trigger the import of parentchain blocks.
+pub trait TriggerParentchainBlockImport<SignedBlockType> {
+	/// Trigger the import of all queued block, **including** the latest one.
+	///
+	/// Returns the latest imported block (if any).
+	fn import_all(&self) -> Result<Option<SignedBlockType>>;
+
+	/// Trigger import of all queued blocks, **except** the latest one.
+	fn import_all_but_latest(&self) -> Result<()>;
+
+	/// Trigger import of all blocks up to **and including** a specific block.
+	///
+	/// If no block in the queue matches, then no blocks will be imported.
+	/// Returns the latest imported block (if any).
+	fn import_until<MatchingF>(&self, matching_func: MatchingF) -> Result<Option<SignedBlockType>>
+	where
+		MatchingF: Fn(&SignedBlockType) -> bool;
+}
```

Comment:
New trait to control when the parentchain blocks are imported. Any blocks that are passed into the dispatcher are queued until an import is triggered through this trait.